src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1026 @@
+#include <stdio.h>
+#include <gurt/common.h>
+#include <hdf5.h>
+#include <daos.h>
+#include <daos_cont.h>
+
+#define ATTR_DSET "User Attributes"
+
+/* for user attr dataset */
+struct usr_attr {
+	char	*attr_name;
+	hvl_t	attr_val;
+};
+
+static int
+serialize_acl(hid_t file_id, struct daos_prop_entry *entry,
+	      const char *prop_str)
+{
+
+	int		rc = 0;
+	int		i = 0;
+	hid_t		status = 0;
+	struct daos_acl	*acl = NULL;
+	char		**acl_strs = NULL;
+	size_t		len_acl = 0;
+	hsize_t		attr_dims[1];
+	hid_t		attr_dtype = 0;
+	hid_t		attr_dspace = 0;
+	hid_t		usr_attr = 0;
+
+
+	if (entry == NULL || entry->dpe_val_ptr == NULL) {
+		D_GOTO(out, rc);
+	}
+
+	/* convert acl to list of strings */
+	acl = (struct daos_acl *)entry->dpe_val_ptr;
+	rc = daos_acl_to_strs(acl, &acl_strs, &len_acl);
+	if (rc != 0) {
+		D_ERROR("failed to convert acl to strs "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	attr_dims[0] = len_acl;
+	attr_dtype = H5Tcopy(H5T_C_S1);
+	if (attr_dtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to create acl type "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
```

Comment:
@daltonbohning not really that I know of....HDF5 just returns a negative value if unsuccessful. I removed printing the -DER_MISC errors but still return it, but no not sure the "best practice" in this case. I will do it this way without printing unless someone has a better suggestion.

Reply:
@daltonbohning well actually there may be a way to get more descriptive error codes from HDF5: https://support.hdfgroup.org/ftp/HDF/prev-Documentation/HDF4.1r3/Users_Guide/UG41r3_html/Errors.fm2.html

Reply:
@dsikich Honestly, I'm not sure. But what I was thinking was something like this:
```
if (attr_dtype < 0) {
	D_ERROR("failed to create acl type\n");
	D_GOTO(out, rc = -DER_MISC);
}
```
So, print SOME error, but just not necessarily the -DER_MISC, since it isn't really helpful anyway. But I'm not sure what the best approach is here.
It's also possible that maybe instead of printing an error here (and the other similar calls in this function), we can log them as DEBUG messages? Maybe we should compare to other places in DAOS.

Reply:
@dsikich Maybe at some point (future?) we can created a wrapper/define for converting the HDF errors to a relevant DAOS error? Or at least printing something reasonable